src/ui/foundation/MainLayout/data.tsx

```diff
@@ -8,73 +8,65 @@ import {
 } from '@mui/icons-material';
 import { TreeItem, TreeView } from '@mui/lab';
 import { IconButton, TextField, Typography } from '@mui/material';
-import { NavLink, useNavigate } from 'react-router-dom';
+import { NavLink, useNavigate, NavigateFunction } from 'react-router-dom';
+import React from 'react';
 
 import { pathFor, PathName } from '@/ui/utilities/paths';
+import { useVaultPages } from '@/ui/hooks/query/useVaultPages';
+import { parsePathListToTree, TreeNode } from '@/lib/path/pathListToTree';
+
+function createTreeItem(
+  node: TreeNode,
+  navigateFunction: NavigateFunction
+): React.ReactNode {
+  const children: React.ReactNode[] = [];
+
+  node.children.forEach((child) => {
+    children.push(createTreeItem(child, navigateFunction));
+  });
+
+  let onClick;
+  if (node.children.length === 0) {
+    onClick = () => {
+      navigateFunction(
+        pathFor({
+          path: PathName.PAGES,
+          params: { filepath: node.fullName },
+        })
+      );
+    };
+  }
```

Comment:
You should avoid mutating "regular" variables in react component, and instead favor states or functional functions. In this case, a functional component would be what you want, eg:

```typescript
const handleOnClick = () => {
  if (node.children.length === 0) {
    navigateFunction(
      pathFor({
        path: PathName.PAGES,
        params: { filepath: node.fullName },
      })
    );
  }
}
```

Then 

```typescript
  ..
  onClick={handleOnClick}
```